standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DatabaseProduct.java

```diff
@@ -203,7 +203,7 @@ public boolean isTableNotExistsError(SQLException e) {
    * Whether the RDBMS has restrictions on IN list size (explicit, or poor perf-based).
    */
   protected boolean needsInBatching() {
-    return isORACLE() || isSQLSERVER();
+    return isORACLE() || isSQLSERVER() || isPOSTGRES();
```

Comment:
Setting hive.metastore.direct.sql.batch.size should have the same effect as this fix. We can just ask the customer to set this configuration variable instead of needing this fix.

This was the reason I did not submit a PR with this fix in the first place.